src/pyhf/cli/infer.py

```diff
@@ -17,6 +18,114 @@ def cli():
     """Infererence CLI group."""
 
 
+@cli.command()
+@click.argument("workspace", default="-")
+@click.option(
+    "--output-file",
+    help="The location of the output json file. If not specified, prints to screen.",
+    default=None,
+)
+@click.option("--measurement", default=None)
+@click.option("--value", default=False, is_flag=True)
+@click.option("-p", "--patch", multiple=True)
+@click.option(
+    "--backend",
+    type=click.Choice(["numpy", "pytorch", "tensorflow", "jax", "np", "torch", "tf"]),
+    help="The tensor backend used for the calculation.",
+    default="numpy",
+)
+@click.option(
+    "--optimizer",
+    type=click.Choice(["scipy", "minuit"]),
+    help="The optimizer used for the calculation.",
+    default="scipy",
+)
+@click.option("--optconf", type=EqDelimStringParamType(), multiple=True)
+def fit(
+    workspace,
+    output_file,
+    measurement,
+    value,
+    patch,
+    backend,
+    optimizer,
+    optconf,
+):
+    """
+    Perforam a maximum likelihood fit for a given pyhf workspace.
+
+    Example:
+
+    .. code-block:: shell
+
+        $ curl -sL https://git.io/JJYDE | pyhf fit --value
+
+        \b
+        {
+            "mle_parameters": {
+                "mu": [
+                    0.00017298628839781602
+                ],
+                "uncorr_bkguncrt": [
+                    1.0000015671710816,
+                    0.9999665895859197
+                ]
+            },
+            "twice_nll": 23.19636590468879
+        }
+    """
+    # set the backend if not NumPy
+    if backend in ["pytorch", "torch"]:
+        set_backend("pytorch", precision="64b")
+    elif backend in ["tensorflow", "tf"]:
+        set_backend("tensorflow", precision="64b")
+    elif backend in ["jax"]:
+        set_backend("jax")
+    tensorlib, _ = get_backend()
+
+    optconf = {k: v for item in optconf for k, v in item.items()}
+
+    # set the new optimizer
+    if optimizer:
+        new_optimizer = getattr(optimize, optimizer) or getattr(
+            optimize, f"{optimizer}_optimizer"
+        )
+        set_backend(tensorlib, new_optimizer(**optconf))
+
+    with click.open_file(workspace, "r") as specstream:
+        spec = json.load(specstream)
+    ws = Workspace(spec)
+    patches = [json.loads(click.open_file(pfile, "r").read()) for pfile in patch]
+
+    model = ws.model(
+        measurement_name=measurement,
+        patches=patches,
+        modifier_settings={
+            "normsys": {"interpcode": "code4"},
+            "histosys": {"interpcode": "code4p"},
+        },
+    )
+    data = ws.data(model)
+
+    fit_result = mle_fit(data, model, return_fitted_val=value)
+
+    _pars = fit_result if not value else fit_result[0]
+    bestfit_pars = {
+        k: tensorlib.tolist(_pars[v["slice"]]) for k, v in model.config.par_map.items()
+    }
+
+    result = {"mle_parameters": bestfit_pars}
+    if value:
+        result["twice_nll"] = tensorlib.tolist(fit_result[-1])
```

Comment:
"twice_nll" would change to "nll" in the other PR I have open, but should still leave it for now.

Reply:
ah ok